### DISCUSSION TITLE
Set and pass environment variable from Fastlane to calling bash script

### DISCUSSION THREAD

**Discussion by dbarros** [fastlane/fastlane#21923] [Q&A]
I am wanting to set an environment variable in Fastlane, and pass it back to a calling Bash script.

Just as a simple example, I run this Bash script:

```
#!/bin/sh

fastlane beta
echo $DELIVER_USER
```

Somewhere in my lane code I have:

```
sh(command: 'export DELIVER_USER=joe')
```

But the `echo` in the calling script does not show "joe", it doesn't show anything. Is there a way that I can do this? Or some other way to be able to pass a value from Fastlane back to the calling Bash script?

**Comment by omawnakw** [comment]
Hmm indeed it's weird it doesn't work out of the box but it seems to be a Ruby limitation.
https://stackoverflow.com/questions/2660571/exporting-an-environment-variable-in-ruby#:~:text=You%20can't%20export%20environment,a%20source%2Dable%20bash%20file.
